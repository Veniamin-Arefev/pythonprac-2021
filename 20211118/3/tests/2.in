import random
from string import ascii_lowercase

def timer(func):
    import time
    def timed(*args, **kw):
        start = time.time()
        result = func(*args, **kw)
        return time.time() - start
    return timed

alp = Alpha(c=10, z=2, a=42)
alq = AlphaQ(c=10, z=2, a=42)

@timer
def work(alpha, count, seed=42):
    random.seed(seed)
    for i in range(count):
        setattr(alpha, random.choice(ascii_lowercase), random.randint(228, 1337 + 100500))

iter = 100000
a = work(alp, iter)
b = work(alq, iter)
print(a < b)
