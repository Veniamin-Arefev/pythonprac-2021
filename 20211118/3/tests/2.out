work worked for 0.11000 seconds
work worked for 0.11200 seconds
